x-pack/plugins/actions/server/monitoring/register_cluster_collector.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import stats from 'stats-lite';
+import { MonitoringCollectionSetup } from '../../../monitoring_collection/server';
+import { CoreSetup } from '../../../../../src/core/server';
+import { ActionsPluginsStart } from '../plugin';
+import { ClusterActionsMetric } from './types';
+
+export function registerClusterCollector({
+  monitoringCollection,
+  core,
+}: {
+  monitoringCollection: MonitoringCollectionSetup;
+  core: CoreSetup<ActionsPluginsStart, unknown>;
+}) {
+  monitoringCollection.registerMetric({
+    type: 'cluster_actions',
+    schema: {
+      overdue: {
+        count: {
+          type: 'long',
+        },
+        duration: {
+          p50: {
+            type: 'long',
+          },
+          p99: {
+            type: 'long',
+          },
+        },
+      },
+    },
+    fetch: async () => {
+      const services = await core.getStartServices();
+      const nowInMs = +new Date();
+      const { docs: overdueTasks } = await services[1].taskManager.fetch({
+        query: {
```

Comment:
Is there any way we can export and reuse the query that's used in task manager so we can make sure they stay in sync? 

https://github.com/elastic/kibana/blob/1297d7da56ff54d01b6e5ac2710eb6b05619a362/x-pack/plugins/task_manager/server/queries/task_claiming.ts#L386-L390

Reply:
Fantastic idea!